lib/baseTest.py

```diff
@@ -132,21 +136,39 @@ def remove_test_file(self):
         if hasattr(self, "test_url_id"):
             self.target_helper.delete_target(self.test_url_id)
 
-    def load_configs(self):
+    def _platform_dep_settings_handler(self, config_value):
+        if self.current_platform_name in config_value:
+            if self.current_platform_ver in config_value[self.current_platform_name]:
+                platform_dep_variables = copy.deepcopy(config_value[self.current_platform_name][self.current_platform_ver])
+            else:
+                platform_dep_variables = copy.deepcopy(config_value[self.current_platform_name][''])
+            config_value.update(platform_dep_variables)
+            return platform_dep_variables
+        return {}
+
+    def set_configs(self, config_variable_name, config_value):
         default_platform_dep_settings_key = "platform-dep-settings"
+        if default_platform_dep_settings_key in config_value:
+            # Load the index config's settings under "platform-dep-settings" base on platform
+            platform_dep_variables = self._platform_dep_settings_handler(config_value[default_platform_dep_settings_key])
+            config_value.update(platform_dep_variables)
+            config_value.pop(default_platform_dep_settings_key)
+
+        if hasattr(self, config_variable_name):
+            # getattr is a way to get variable by reference and doesn't need to be set back
+            getattr(self, config_variable_name).update(config_value)
```

Comment:
I think we should check the accept config name.
And I will suggest modify it to two line for readable reason.